### PR TITLE
Stm32f746 disco

### PR DESCRIPTION
**Pull request by jinsheng20**
## 拉取/合并请求描述：(PR description)

增加以太网和sdram
已经通过mdk,iar,gcc软件，在stm32f746-disco开发板上验证。

以下的内容不应该在提交PR时的message修改，修改下述message，PR会被直接关闭。请在提交PR后，浏览器查看PR并对以下检查项逐项check，没问题后逐条在页面上打钩。
The following content must not be changed in submitted PR message. Otherwise, the PR will be closed immediately. After submitted PR, please use web browser to visit PR, and check items one by one, and ticked them if no problem.

### 当前拉取/合并请求的状态 Intent for your PR

必须选择一项 Choose one (Mandatory):

- [ ] 本拉取/合并请求是一个草稿版本 This PR is for a code-review and is intended to get feedback
- [x] 本拉取/合并请求是一个成熟版本 This PR is mature, and ready to be integrated into the repo

### 代码质量 Code Quality：

我在这个拉取/合并请求中已经考虑了 As part of this pull request, I've considered the following:

- [x] 已经仔细查看过代码改动的对比 Already check the difference between PR and old code
- [x] 代码风格正确，包括缩进空格，命名及其他风格 Style guide is adhered to, including spacing, naming and other style
- [x] 没有垃圾代码，代码尽量精简，不包含`#if 0`代码，不包含已经被注释了的代码 All redundant code is removed and cleaned up
- [x] 所有变更均有原因及合理的，并且不会影响到其他软件组件代码或BSP All modifications are justified and not affect other components or BSP
- [x] 对难懂代码均提供对应的注释 I've commented appropriately where code is tricky
- [x] 本拉取/合并请求代码是高质量的 Code in this PR is of high quality

### CODE REVIEW

File: bsp/stm32/stm32f746-st-disco/board/ports/phy_reset.c [hunk 1]
```diff
@@ -0,0 +1,24 @@
+/*
+ * Copyright (c) 2006-2018, RT-Thread Development Team
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * Change Logs:
+ * Date           Author       Notes
+ * 2019-02-16     jinsheng   first version
```
Comment (SummerGGift) [inline bsp/stm32/stm32f746-st-disco/board/ports/phy_reset.c]: 格式对齐

File: bsp/stm32/stm32f746-st-disco/board/ports/sdcard_port.c [hunk 2]
```diff
@@ -0,0 +1,65 @@
+/*
+ * Copyright (c) 2006-2018, RT-Thread Development Team
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * Change Logs:
+ * Date           Author       Notes
+ * 2019-02-17     jinsheng   add sdcard port file
```
Comment (SummerGGift) [inline bsp/stm32/stm32f746-st-disco/board/ports/sdcard_port.c]: 格式对齐

File: bsp/stm32/stm32f746-st-disco/board/ports/sdram_port.h [hunk 3]
```diff
@@ -0,0 +1,64 @@
+/*
+ * Copyright (c) 2006-2018, RT-Thread Development Team
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * Change Logs:
+ * Date           Author       Notes
+ * 2019-02-16     jinsheng   The first version for STM32F7xx
```
Comment (SummerGGift) [inline bsp/stm32/stm32f746-st-disco/board/ports/sdram_port.h]: 格式

File: bsp/stm32/stm32f746-st-disco/board/ports/phy_reset.c [hunk 4]
```diff
@@ -0,0 +1,24 @@
...
+#include <rtthread.h>
+
+//#define DRV_DEBUG
+#define LOG_TAG             "drv.phy_reset"
+#include <drv_log.h>
+
+//PHY RESET PIN link system reset
+
+/* phy reset */
+void phy_reset(void)
```
Comment (misonyo) [inline bsp/stm32/stm32f746-st-disco/board/ports/phy_reset.c]: 函数怎么为空？

Comment (jinsheng20) [reply bsp/stm32/stm32f746-st-disco/board/ports/phy_reset.c]: reset 引脚连接的是整个板载的reset，所以这里就是空

**Pull request merged**